mlir/lib/Dialect/Bufferization/IR/BufferizationOps.cpp

```diff
@@ -73,18 +73,23 @@ mlir::bufferization::castOrReallocMemRefValue(OpBuilder &b, Value value,
     Value size = b.create<memref::DimOp>(loc, value, i);
     dynamicOperands.push_back(size);
   }
-  // TODO: Use alloc/memcpy callback from BufferizationOptions if called via
-  // BufferizableOpInterface impl of ToMemrefOp.
-  Value copy = b.create<memref::AllocOp>(loc, destType, dynamicOperands);
-  b.create<memref::CopyOp>(loc, value, copy);
+
+  FailureOr<Value> copy =
+      options.createAlloc(b, loc, destType, dynamicOperands);
```

Comment:
I noticed in the tests that this call results in an aligned allocation which was not the case before (e.g. when lowering `to_memref(to_tensor(x))`). May this cause problems anywhere?

Reply:
It's a functional change, but I don't think it will be problematic. Mainly because other folds used by bufferization already use `options.createAlloc`.